napari/_vispy/vispy_image_layer.py

```diff
@@ -43,11 +52,22 @@ def _on_display_change(self):
         self.reset()
 
     def _on_data_change(self):
+        data = self.layer._data_view
+        dtype = np.dtype(data.dtype)
+        if dtype not in texture_dtypes:
+            try:
+                dtype = dict(i=np.int16, f=np.float64)[dtype.kind]
```

Comment:
Can you explain this line?

Reply:
@jni this is equivalent to
```python
if dtype.kind == 'i':    # is an int
    dtype = np.int16
elif dtype.kind == 'f':  # is a float
    dtype = np.float64
else:
    raise TypeError(f'type {dtype} not allowed for texture; must be one of {set(texture_dtypes)}')
```

Reply:
Ok, but why is np.int6 the default int type?

Reply:
@sofroniewn ^

Reply:
also see [here](https://github.com/vispy/vispy/blob/maint/0.6/vispy/gloo/glir.py#L1409): `np.float64` is not a valid option at this time

would recommend you directly set `texture_dtypes = vispy.gloo.glir.GlirTexture._types.keys()` [as suggested in my post](https://github.com/napari/napari/issues/447#issuecomment-522079016)

bonus points if you don't hardcode the default and figure them out based on `texture_dtypes` ;)

Reply:
I stared with your recommendation, but found that list diverged from what threw errors when I tried running things. `np.float64` ran fine, but wasn't on that list. `np.int32` failed despite being on that list, which is what lead me to the hard coding approach that I took. maybe i got something wrong when trying them, but that was my experience

Reply:
Just a note: I would avoid anything else than uint8, uint16, float32 for vispy/OpenGL textures. Support for the other types such as 16bit and 64bit types might be spotty or not fully supported in all machines. Also, 64 bit textures is overkill and a waste of precious GPU memory, so I would avoid that anyway -- particularly for 3D textures (volume rendering).

Reply:
ok now `float64` will be converted to `float32` before passing to vispy